modules/internal/ExternalArray.chpl

```diff
@@ -96,34 +96,72 @@ module ExternalArray {
     return ret;
   }
 
+  proc convertStringOrBytes(ref arr: []): chpl_external_array
+    where arr.eltType == string || arr.eltType == bytes {
+
+    var wrapper: [arr.domain] chpl_byte_buffer;
+
+    // Move existing string buffers over to a new shell.
+    for i in 0..#arr.size {
+      ref item = arr[i];
+
+      // This call assumes ownership of the string's buffer.
+      var val = chpl__exportRetStringOrBytes(item);
+      wrapper[i] = val;
+    }
+
+    var ext = chpl_make_external_array_ptr_free(c_ptrTo(wrapper[0]),
+                                                wrapper.size: uint);
+
+    return ext;
+  }
+
   proc convertToExternalArray(in arr: []): chpl_external_array
-    where (getExternalArrayType(arr) == chpl_external_array) {
-    if (!isExternArrEltType(arr.eltType)) {
+    where getExternalArrayType(arr) == chpl_external_array {
+
+    if !isExternArrEltType(arr.eltType) {
```

Comment:
The parentheses removal seems unnecessary

Reply:
As do many of the conversions to `if ... then` XD

Reply:
Sure.